### DISCUSSION TITLE
Missing Kafka Server Traces in OpenTelemetry Setup

### DISCUSSION THREAD

**Discussion by jiten686** [open-telemetry/opentelemetry-java-instrumentation#12448] [Q&A]
I have Service-A which produced the data to Kafka and Service-B consume the data. I have integrated open telemetry java agent-2.6 to all services. (i.e. Service-A, Service-B and Kafka Server), However, I am seeing the traces for Service-A and Service-B only (i.e Service-A -> Service-B) with same trace-id , Why I am not able to see the Traces for Kafka-server (i.e. Service-A -> Kafka -> Service-B) ? 
Is it expected to see the traces for kafka-server with same trace-id ? 

What additional changes do I need to make?

**Comment by laurit** [comment]
We don't provide instrumentation for kafka server.

**Reply by jiten686** [reply]
Is there any plan in future to provide instrumentation for Kafka-server?

**Reply by laurit** [reply]
For new instrumentations we rely on the community to contribute them. I'm not aware of anybody working on trying to instrument the kafka server.
